pkg/storage/edv/encryptedformatter.go

```diff
@@ -167,15 +161,9 @@ func (f *EncryptedFormatter) getStructuredDocFromEncryptedDoc(
 	return structuredDocument, nil
 }
 
-type generateRandomBytesFunc func([]byte) (int, error)
-
-func generateEDVCompatibleID(generateRandomBytes generateRandomBytesFunc) (string, error) {
-	randomBytes := make([]byte, 16)
-
-	_, err := generateRandomBytes(randomBytes)
-	if err != nil {
-		return "", err
-	}
+// GenerateEDVCompatibleID generate edv compatible id.
+func (f *EncryptedFormatter) GenerateEDVCompatibleID(k string) string {
+	hashKey := sha256.New().Sum([]byte(k))
```

Comment:
Use HMAC function to prevent EDV server from guessing the `k` and leave a ToDo to investigate this "random ID" issue later.

EDV server will accept the ID still as long as it's in the right format (Base58 and was 128 bits before being encoded as Base58)

(you might have to convert that HMAC hash to be the right length)